test/message-queues.ts

```diff
@@ -209,6 +210,40 @@ describe('MessageQueues', () => {
         setImmediate(() => messageQueue.flush());
         return promise;
       });
+
+      it('should resolve onDrain only after all in-flight messages have been flushed', async () => {
+        const patchedQueue = new MessageQueue(subscriber);
+        const origSendBatch = patchedQueue._sendBatch;
+        const log: string[] = [];
+        const sendDone = defer();
+        patchedQueue._sendBatch = async function _sendBatch(
```

Comment:
I wonder if this might be done using `sandbox.stub` just to be consistent with the rest of the tests. I'm going to ask someone who knows a bit more about Sinon than me.

Reply:
Do you mean just the patching of the method?

Reply:
Yep! Sorry for not being clear there. On second thought, I think it's not a big deal because you're making your own `MessageQueue` anyway.

Reply:
I wanted to stay consistent with the rest of the source but I was not sure how else I would pull this off. Do you have any suggestions?

Reply:
I think you could probably use the sandbox on the existing `messageQueue._sendBatch`, if you wanted to be a bit more consistent with the other tests. That might be nice if we change something about how the "global" `MessageQueue` is wired up under `describe` later, but I don't think it's worth spending much time on.

Reply:
I changed it and rebased to tip of master but now other unrelated tests are failing. Is master healthy right now?

Reply:
It _should_ be, but we have been tracking down some flaky tests for a bit, too. I'll look.